neurom/features/neurite_features.py

```diff
@@ -109,6 +110,19 @@ def number_of_sections_per_neurite(neurites, neurite_type=TreeType.all):
     return (_sec.count(n) for n in neurites if _ttc(neurite_type)(n))
 
 
+@as_neurite_list
+def total_length_per_neurite(neurites, neurite_type=TreeType.all):
+    '''Get an iterable with the total length of a neurite for a given neurite type'''
+    return (sum(_sec.length(ss) for ss in isection(n))
```

Comment:
Yes, something like that. Although, as I said, I realise that might be a bit awkward in python. But it might be an idea to try it that way first, and if it is too complicated or the one-liner becomes too complex, then we can use the functions (and I can see if I can simplify the design.)